### PR TITLE
PARQUET-753: Fixed GroupType.union() to handle original type

### PR DESCRIPTION
**Pull request by adeneche**
also fixed GroupType.equals() to compare the original type and 2 unit tests that weren't setting the original type properly on the expected results

### CODE REVIEW

**Comment by julienledem** [comment]
LGTM. Please open a Parquet JIRA

**Comment by adeneche** [comment]
opened PARQUET-753

**Comment by liancheng** [comment]
+1

**Comment by julienledem** [comment]
@rdblue is this one ok to merge or do you prefer to wait after the release?

**Comment by rdblue** [comment]
I think it's fine

File: parquet-column/src/main/java/org/apache/parquet/schema/GroupType.java [hunk 1]
```diff
@@ -1,14 +1,14 @@
-/*
```
Comment (adeneche) [inline parquet-column/src/main/java/org/apache/parquet/schema/GroupType.java]: my IDE is probably messing with those, I will fix this

**Pull request closed without merging**